### PR TITLE
use the "None" datasource for on-first-boot configuration

### PR DESCRIPTION
**Pull request by mwhudson**
Using the NoCloud source meant that a filesystem label of "cidata"
(probably containing autoinstall config) could override the cloud-init
nocloud seed subiquity wrote and then users would not get created. So
instead write cloud config directly that hardcodes using the
DataSourceNone source and config for it.

https://bugs.launchpad.net/subiquity/+bug/1879103

### CODE REVIEW

**Comment by mwhudson** [comment]
Would definitely like review from people who understand cloud-init better than me :)

**Comment by xnox** [comment]
I think there was a way to specify in the ds-identify configuration _which_ explicit path of NoNet one wants to use.

I.e. I just hope this current implementation doesn't cause cloud-init to attempt searching all the datasources on the internet.

**Comment by xnox** [comment]
Plus if anybody specified their datasource _after_ `---` on the kernel cmdline during install, that propagates to the installed system, and must be removed too.

Because kernel commandline overrides anything we encode on disk.

File: subiquity/models/subiquity.py [hunk 1]
```diff
@@ -203,10 +203,18 @@ def _cloud_init_files(self):
...
+                "None": {
+                    'userdata_raw': userdata,
+                    'metadata': metadata,
+                    },
+                },
+            }
         files = [
-            ('var/lib/cloud/seed/nocloud-net/meta-data', metadata, 0o644),
-            ('var/lib/cloud/seed/nocloud-net/user-data', userdata, 0o600),
+            ('etc/cloud/cloud.cfg.d/installer.cfg', yaml.dump(config), 0o600),
```
Comment (raharper) [inline subiquity/models/subiquity.py:217]: I would change this to 99-subiquity.cfg;  using number > 90 will ensure this clobers the built-in 90-dpkg.cfg and I think including subiquity is useful to know what program wrote this (also consider a # comment in the yaml with the signature as well.

Comment (mwhudson) [reply subiquity/models/subiquity.py:217]: That makes sense, I didn't realise that that's what 90_dpkg.cfg had in it. I'm a bit confused as to how this branch helps with the cloud.cfg.d snippets being in the order they are, but it does...

Comment (raharper) [reply subiquity/models/subiquity.py:217]: This is likely due to luck/accident.  ds-identify does not have a full yaml parser, but rather searches for keys while globbing through /etc/cloud/cloud.cfg /etc/cloud/cloud.cfg.d/*.cfg;

Comment (mwhudson) [reply subiquity/models/subiquity.py:217]: Oh yay. I've updated the filename.

**Comment by mwhudson** [comment]
I'll merge this soon -- it may not be perfect but it's definitely an improvement.

**Pull request merged**